core/src/main/java/org/lflang/federated/generator/FederateInstance.java

```diff
@@ -713,7 +735,8 @@ public TimeValue findNearestPhysicalActionTrigger(ReactionInstance reaction) {
         // Outputs of contained reactions
         PortInstance outputInstance = (PortInstance) trigger;
         for (ReactionInstance uReaction : outputInstance.getDependsOnReactions()) {
-          TimeValue uMinDelay = findNearestPhysicalActionTrigger(uReaction);
+          visited.add(uReaction); // Mark the upstream reaction as visited.
```

Comment:
Is this add premature? Won't the recursive call on the next line add it if it isn't there already?  And if added here, won't the recursive call actually not do anything?

Reply:
Yes, this is a good point. I've moved this to the front of the method.